driver-sync/src/test/functional/com/mongodb/client/unified/Entities.java

```diff
@@ -480,6 +496,41 @@ private void initBucket(final BsonDocument entity, final String id, final Functi
         putEntity(id, gridFSBucketSupplier.apply(database), buckets);
     }
 
+    private void initClientEncryption(final BsonDocument entity, final String id) {
+        assumeTrue(hasEncryptionTestsEnabled());
+        assumeTrue(serverVersionAtLeast(4, 2));
+
+        if (!entity.containsKey("clientEncryptionOpts")) {
+            throw new UnsupportedOperationException("Unsupported client encryption specification missing: clientEncryptionOpts");
+        }
+        BsonDocument clientEncryptionOpts = entity.getDocument("clientEncryptionOpts");
+        if (!clientEncryptionOpts.containsKey("keyVaultClient")) {
+            throw new UnsupportedOperationException("Unsupported client encryption specification missing: "
+                    + "clientEncryptionOpts.keyVaultClient");
+        }
+
+        MongoClient mongoClient = null;
+        ClientEncryptionSettings.Builder builder = ClientEncryptionSettings.builder();
+        for (Map.Entry<String, BsonValue> entry : clientEncryptionOpts.entrySet()) {
+            switch (entry.getKey()) {
+                case "keyVaultClient":
+                    mongoClient = clients.get(entry.getValue().asString().getValue());
+                    break;
+                case "keyVaultNamespace":
+                    builder.keyVaultNamespace(entry.getValue().asString().getValue());
+                    break;
+                case "kmsProviders":
+                    builder.kmsProviders(createKmsProvidersMap(entry.getValue().asDocument()));
+                    break;
+                default:
+                    throw new UnsupportedOperationException("Unsupported client encryption option: " + entry.getKey());
+            }
+        }
+
+        ClientEncryption clientEncryption = new ClientEncryptionImpl(Assertions.notNull("mongoClient", mongoClient), builder.build());
```

Comment:
This needs an indirection similar to what is done in `Entities#init`, so that when testing reactive streams we make use of `com.mongodb.reactivestreams.client.syncadapter.SyncClientEncryption`

Reply:
Great catch.